[NextGenDMImageURIBuilder] Non-deterministic behavior of `HashMap` might fail tests

<!-- 

Before sending a bug report please check if the latest release (if you're not already using it) fixes your problem. Alternatively, you could
install the latest snapshot version from the master branch.
 -->

## Bug Report

**Current Behavior**
- `NextGenDMImageURIBuilder` builds the NextGen Dynamic Media Image URI with different parameters using `HashMap` at https://github.com/adobe/aem-core-wcm-components/blob/0efca91f4b7b5f56a02fefabbdec273c8b77ffd8/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/NextGenDMImageURIBuilder.java#L106
https://github.com/adobe/aem-core-wcm-components/blob/0efca91f4b7b5f56a02fefabbdec273c8b77ffd8/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/NextGenDMImageURIBuilder.java#L123-L131
- However, the official Javadoc of [HashMap](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html) and [HashSet](https://docs.oracle.com/javase/8/docs/api/java/util/HashSet.html), which is internally used by the `HashMap`, mentions that the order or elements will not be maintained over time.
- This causes issue where the URI String with params are expected to be in a specific order. Specifically, the following tests expects the order of params in the generated URI to be in their order of insertion.
  - `com.adobe.cq.wcm.core.components.internal.models.v3.NextgenDMImageURIBuilderTest#testUriWithDefaultValues`
  - `com.adobe.cq.wcm.core.components.internal.models.v3.NextgenDMImageURIBuilderTest#testUriWithInvalidSmartCrop`
  - `com.adobe.cq.wcm.core.components.internal.models.v3.ImageImplTest#testNgdmImage`
  - `com.adobe.cq.wcm.core.components.internal.models.v3.ImageImplTest#testNgdmImageWithResizeWidth`
  - `com.adobe.cq.wcm.core.components.internal.models.v2.TeaserImplTest#testTeaserWithNgdmImage`
- However, this expected order might be changed based on the order of elements returned by `param.entrySet()`, which could result in the following error. To be concise, I have mentioned only one of the error logs.
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ImageImplTest.testNgdmImageWithResizeWidth:690 \
expected: <{"id":"image-042059c802","src":"https://testrepo/adobe/dynamicmedia/deliver/urn:aaid:aem:e82c3c87-1453-48f5-844b-1822fb610911/cutfruits.png?width=800&preferwebp=true&crop=2:3,smart","lazyEnabled":true,"sizes":"","dataLayer":{"image-042059c802":{"@type":"core/wcm/components/image/v3/image"}},":type":"core/wcm/components/image/v3/image"}> \
but was: <{"id":"image-042059c802","src":"https://testrepo/adobe/dynamicmedia/deliver/urn:aaid:aem:e82c3c87-1453-48f5-844b-1822fb610911/cutfruits.png?preferwebp=true&width=800&crop=2:3,smart","sizes":"","lazyEnabled":true,"dataLayer":{"image-042059c802":{"@type":"core/wcm/components/image/v3/image"}},":type":"core/wcm/components/image/v3/image"}>
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
```
- Extracting just the params from the above logs
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ImageImplTest.testNgdmImageWithResizeWidth:690 
expected: ...cutfruits.png?width=800&preferwebp=true&crop=2:3,smart... 
but was: ...cutfruits.png?preferwebp=true&width=800&crop=2:3,smart...
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
```
- The exact URI error varies based on the params used by the tests. However, the root cause remains the same.

#### Steps to reproduce
- We could observe the above mentioned issue by running the test through [NonDex](https://github.com/TestingResearchIllinois/NonDex), which shuffles the order of elements returned by `params.entrySet()`
```
# Clone the Repo
https://github.com/adobe/aem-core-wcm-components.git

# Compile the module byte-buddy-dep
mvn clean install -pl bundles/core -am -DskipTests

# (Optional) Run the unit test
mvn -pl bundles/core test -Dtest=<Test Name>

# Run the unit test using NonDex
mvn -pl bundles/core edu.illinois:nondex-maven-plugin:2.1.1:nondex -Dtest=<Test Name>
```

**Expected behavior/code**
- For any amount of runs, the test should have passes successfully (with `NonDex` too). 
- The expected order of params should be maintained over time. 

**Environment**
- Core Components version: `2.23.5-SNAPSHOT`
- Java version: `Java(TM) SE Runtime Environment (build 1.8.0_381-b09)`

**Possible Solution**
- Since the order of elements is expected to be maintained, replacing the implementation of `params` from `HashMap` to `LinkedHashMap` will enforce the expected order.
```
+/-  com.adobe.cq.wcm.core.components.internal.models.v3.NextGenDMImageURIBuilder.java: 106

-    Map<String, String> params = new HashMap<>();
+    Map<String, String> params = new LinkedHashMap<>();
```
- I am happy to discuss more about the proposed solution. If the proposed solution looks good, I can create a PR.

**Additional context / Screenshots**
- One of the order returned by `params.entrySet()`, which is the expected behavior
<img width="625" alt="Screenshot 2023-11-09 at 12 09 42 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/0cc22d07-3e2b-49a9-ad52-c5d2002f287e">

- The string built based on the above order
<img width="1198" alt="Screenshot 2023-11-09 at 12 09 53 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/dc7bc5ab-9a36-49a7-8227-7b8a5c351f10">

- Another order returned by `params.entrySet()`, which is not expected
<img width="643" alt="Screenshot 2023-11-09 at 12 04 11 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/b4fa9515-878d-4470-876f-b00644e46c43">

- The string built based on the above order
<img width="1198" alt="Screenshot 2023-11-09 at 12 46 14 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/ef8ccbc4-40ea-4b35-a0c5-a3f152867eb5">

--- a/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/NextGenDMImageURIBuilder.java
+++ b/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/NextGenDMImageURIBuilder.java
@@ -15,7 +15,7 @@
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
 package com.adobe.cq.wcm.core.components.internal.models.v3;
 
-import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Scanner;
 
@@ -103,7 +103,7 @@ public class NextGenDMImageURIBuilder {
             imageDeliveryPath = imageDeliveryPath.replace(PATH_PLACEHOLDER_FORMAT, assetExtension);
             String repositoryId = this.config.getRepositoryId();
             StringBuilder uriBuilder = new StringBuilder("https://" + repositoryId + imageDeliveryPath);
-            Map<String, String> params = new HashMap<>();
+            Map<String, String> params = new LinkedHashMap<>();
             if(this.width > 0) {
                 params.put("width", Integer.toString(this.width));
             }

